preemptive_requests.py

```diff
@@ -35,23 +77,362 @@ def setup_requests(self):
+        filters in the ui.
+
+        Currently supports the following filters, which are all active by
+        default:
+        - Hide if requesting full beam
+        - Hide if no activate arbitration
+        - Hide if PV disconnected
+
+        Parameters
+        ----------
+        row : int
+            The row of the table to show or hide.
+        """
+        # Treat the entire row as one entity
+        table = self.ui.reqs_table_widget
+        values = {}
+        for info, column in zip(item_info_list, range(2, table.columnCount())):
+            item = table.item(row, column)
+            values[info.name] = item.get_value()
+
+        full_beam = all((
+            values['rate'] >= 120,
+            values['trans'] >= 1,
+            values['energy'] >= 32,
+            ))
+        active = bool(values['active'])
+        connected = item.connected
+
+        hide_full_beam = self.ui.full_beam.isChecked()
+        hide_inactive = self.ui.inactive.isChecked()
+        hide_disconnected = self.ui.disconnected.isChecked()
+
+        hide = any((
+            hide_full_beam and full_beam,
+            hide_inactive and not active,
+            hide_disconnected and not connected,
+            ))
+
+        if hide:
+            table.hideRow(row)
+        else:
+            table.showRow(row)
+
+    def update_all_filters(self, *args, **kwargs):
+        """Call update_filter on every row of the table."""
+        for row in range(self.row_count):
+            self.update_filter(row)
+
     def ui_filename(self):
         return 'preemptive_requests.ui'
+
+    def channels(self):
+        """
+        Make sure PyDM can find the channels we set up for cleanup.
+
+        Put this here instead of on the PMPSTableWidgetItem because
+        QTableWidgetItem instances are not instances of QWidget, and
+        therefore are not checked by PyDM for channels.
+        """
+        return self._channels
+
+
+class PMPSTableWidgetItem(QTableWidgetItem):
+    """
+    QTableWidgetItem with extra utilities for the PMPS UI
+
+    Adds the following features:
+    - Fill value and connection state from PyDMChannel
+    - Configurable sorting
+    - Process inputs before storing in the table
+    - Able to sort as a non-str type
+
+    Parameters
+    ----------
+    store_type : callable
+        Function or type to call on the value from the PyDMChannel before
+        storing in the table.
+    data_type : callable
+        Function or type to call on the str stored in the table before
+        comparing with other PMPSTableWidgetItem instances for sorting.
+    default : Any
+        A starting value for the widget item.
+    channel : str, optional
+        PyDM channel address for value and connection updates.
+    """
+    def __init__(self, store_type, data_type, default,
+                 channel=None, parent=None):
+        QTableWidgetItem.__init__(self, parent)
```

Comment:
```suggestion
        super().__init__(parent)
```

Reply:
Good catch, this and the other one must be leftover from Hugo and/or Cristina where they were forced to keep pydm code python2 compatible...